app/com/m3/octoparts/util/OctoMetricsImpl.scala

```diff
@@ -0,0 +1,27 @@
+package com.m3.octoparts.util
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.kenshoo.play.metrics.MetricsImpl
+import play.api.{ Logger, Configuration }
+import play.api.inject.ApplicationLifecycle
+
+import scala.util.control.NonFatal
+
+/**
+ * During tests, we occasionally have cases where the asynchronous shutting down of an application means that
+ * the previous registry in the singleton has not been unregistered before starting the next test.
+ *
+ * This implementation of Metrics overrides onStart to perform onStop cleanups first before proceeding with
+ * other onStart duties
+ */
+class OctoMetricsImpl(lifecycle: ApplicationLifecycle, configuration: Configuration) extends MetricsImpl(lifecycle, configuration) {
+
+  override def onStart(): ObjectMapper = {
```

Comment:
`onStart` with this custom implementation (`OctoMetricsImpl`) will try to remove the metrics from the internal metrics registry map by invoking the parent `onStop` cleanup method before proceeding with normal `onStart` duties; therefore, should be no error thrown; still, a good idea to catch NonFatal errors.

The default implementation of `onStart` in `MetricsImpl` will not try to clean up first; it will therefore throw an error if there are already metrics registered in the global singleton with the same names as the ones that it tries to register (`IllegalArgumentException`)